extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/select/QuerySelectTest.java

```diff
@@ -0,0 +1,71 @@
+package io.quarkus.hibernate.orm.panache.deployment.test.select;
+
+import static java.util.Collections.singletonList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.transaction.Transactional;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.orm.panache.PanacheQuery;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class QuerySelectTest {
+
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource("application.properties")
+                    .addClasses(Person.class));
+
+    @Transactional
+    @Test
+    void testSelect() {
+        Person person1 = new Person();
+        person1.name = "Jo";
+        person1.surname = "Ne";
+        person1.height = 10;
+        Person person2 = new Person();
+        person2.name = "Yep";
+        person2.surname = "Nope";
+        person2.height = 20;
+        Person.persist(person1, person2);
+
+        PanacheQuery<Person> query = Person.find("order by name asc").select(singletonList("name"), result -> {
+            Person p = new Person();
+            p.name = result.toString();
+            return p;
+        });
+        List<Person> names = query.list();
+        assertEquals(2, names.size());
+        assertEquals("Jo", names.get(0).name);
+        assertEquals("Jo", query.firstResult().name);
+
+        List<Integer> heights = Person.find("order by height desc").select(singletonList("height"), result -> (Integer) result)
+                .list();
```

Comment:
So this should just work OOTB without `select` if we drop the type argument constraint:

```java
List<Integer> heights = Person.find("height order by height desc").list();
```

We should drop that type argument constraint.

Reply:
Sounds good. What class should I look at?

Reply:
I think it's the `PanacheQuery` class, or methods. And `list`/`stream`/`find` methods. Note that I haven't tested this, we have to check that the tests still compile if we drop it.

Reply:
Ok, we can fix this in a followup PR.